dapps/v2Explorer/src/screens/App.tsx

```diff
@@ -32,23 +41,86 @@ function App(): JSX.Element {
 
   const getAddress = useCallback(async () => {
     try {
-      const signer = web3Provider.getSigner();
-      const currentAddress = await signer.getAddress();
-      setCurrentAccount(currentAddress);
+      if (web3Provider) {
+        const signer = web3Provider.getSigner();
+        const currentAddress = await signer.getAddress();
+        setCurrentAccount(currentAddress);
+      }
+    } catch (err: unknown) {
+      console.log('Error in getAddress', err);
+    }
+  }, [setCurrentAccount]);
+
+  const onSessionCreated = useCallback(async () => {
+    getAddress();
+    setModalVisible(false);
+  }, [getAddress]);
+
+  const onSessionRejected = useCallback(async () => {
+    setModalVisible(false);
+  }, []);
+
+  const onSessionDelete = useCallback(
+    async ({topic}: {topic: string}) => {
+      if (topic === universalProviderSession?.topic) {
+        clearSession();
+        setCurrentAccount(null);
+      }
+    },
+    [setCurrentAccount],
+  );
+
+  const onConnect = useCallback(async () => {
+    createUniversalProviderSession({
+      onSuccess: onSessionCreated,
+      onFailure: onSessionRejected,
+    });
+    setModalVisible(true);
+  }, [onSessionCreated, onSessionRejected]);
+
+  const onDisconnect = useCallback(async () => {
+    try {
+      await universalProvider.disconnect();
+      clearSession();
+      setCurrentAccount(null);
     } catch (err: unknown) {
-      console.log('Error for initializing', err);
+      Alert.alert('Error', 'Error disconnecting');
     }
   }, []);
 
+  const subscribeToEvents = useCallback(async () => {
```

Comment:
Nitpick: Would it make sense to export this function into `utils/UniversalProvider`?

Reply:
as discussed in slack, we'll keep it there until v1.3